src/Message/Request/PurchaseRequest.php

```diff
@@ -146,6 +146,19 @@ public function getData()
             $data['paymentBrandForce'] = $this->getEnforcePaymentMethod();
         }
 
+        $card = $this->getCard();
+        if (isset($card)) {
+            $data['shippingDetail'] = [
+                'firstName' => (string)$card->getFirstName(),
+                'middleName' => '',
+                'lastName' => (string)$card->getLastName(),
+                'street' => (string)$card->getAddress1(),
+                'postalCode' => (string)$card->getPostcode(),
+                'city' => (string)$card->getCity(),
+                'countryCode' => (string)$card->getCountry(),
```

Comment:
Does it need to be an empty string instead of null? Or why the casts?

Reply:
The documentation states that the country code may not be null. The same for the other fields that are casted.